src/main/java/com/bieganski/ox/checkers/HorizontalWinChecker.java

```diff
@@ -4,45 +4,42 @@
 
 import java.util.TreeSet;
 
-class HorizontalWinChecker implements WinChecker {
+class HorizontalWinChecker extends WinChecker {
+  HorizontalWinChecker(int boardSize, int winSize) {
+    super(boardSize, winSize);
+  }
+
   /**
    * {@inheritDoc}
    */
   @Override
-  public boolean checkWin(TreeSet<Field> fieldsWithValue, Field addedField, int size) {
-    int winSize = 3;
-
+  boolean checkWin(TreeSet<Field> fieldsWithValue, Field addedField) {
+    TreeSetHelper helper = new TreeSetHelper(fieldsWithValue);
     TreeSet<Field> correct = new TreeSet<>();
     correct.add(addedField);
-    Field field;
-    int counter = 0;
-    //TODO DRY and too long
-    while (correct.size() < winSize && counter < 2) {
-      field = correct.last();
-      Field nextField = fieldsWithValue.higher(field);
-      counter = counter + appendFieldIfCorrect(field, nextField, correct);
+    boolean upper = true;
+    boolean lower = true;
 
-      field = correct.first();
-      nextField = fieldsWithValue.lower(field);
-      counter = counter + appendFieldIfCorrect(field, nextField, correct);
+    Field nextField;
+    Field field;
+    while (correct.size() < winSize && !(!upper && !lower)) {
+
+      if (upper) {
+        field = correct.last();
+        nextField = helper.getHigherField(field, field.hashCode() + 1);
+        upper = appendFieldIfCorrect(field, nextField, correct);
+      }
+      if (lower) {
+        field = correct.first();
+        nextField = helper.getLowerField(field, field.hashCode() - 1);
+        lower = appendFieldIfCorrect(field, nextField, correct);
+      }
```

Comment:
This two are very similar, could be extracted to separate method, You can use functional interfaces to do this